server/users.js

```diff
@@ -1568,26 +1571,22 @@ class User extends Chat.MessageContext {
 			this.chatQueue = null;
 		}
 	}
-	isAway() {
-		return this.status && this.status.charAt(0) === '!';
-	}
 	/**
-	 * @param {string} message
+	 * @param {StatusType} type
+	 * @param {string} [message]
 	 */
-	setStatus(message) {
-		if (message === this.status) return;
-		this.status = message;
+	setStatus(type = this.statusType, message) {
```

Comment:
`setStatusType` and `setStatusMessage` should be separate functions; nothing should set both at the same time.